src/components/Input/Input.tsx

```diff
@@ -30,14 +30,19 @@ interface BaseProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 's
     label?: string;
     /** Helper text under the input, display the error when the context is critical */
     helperText?: string;
+    /** Icon before input, optional */
+    leadingIcon?: React.ReactElement;
+    /** Icon after input, which can trigger callback as a button, optional */
+    trailingIcon?: { icon: React.ReactElement; callback: () => void };
     /**  Controls whether space is reserved for error messages under the input field when validation is expected
      *  to avoid "jumping" UI. */
     reserveErrorMessageSpace?: boolean;
 }
 
-export type Props = BaseProps & (ErrorStateProps | { context?: undefined; errorMessage?: never });
+export type Props = BaseProps &
+    (ErrorStateProps | { context?: undefined | string; errorMessage?: undefined | string });
 
-const { block, elem } = bem('Input', styles);
+const { block, elem } = bem('InputContainer', styles);
```

Comment:
I would say to keep the name as Input here and use the block in the upper element, maybe transforming it into a div.